src/charm.py

```diff
@@ -100,6 +100,16 @@ def _on_install(self, event: ops.framework.EventBase) -> None:
         except Exception:
             logger.exception("Failed to hold microceph refresh: ")
 
+    def _on_stop(self, event: ops.StopEvent):
+        """Removes departing unit from the MicroCeph cluster forcefully."""
+        try:
```

Comment:
It would be good to have some logging for the branches of this try/except

Reply:
the microceph methods (remove_cluster_member and is_cluster_member) both log the command+output with `debug` level logs. Since this is an event handler, the entrypoint is always logged and if the except check fails, the exception will always have trace in logs.